Backend/controllers/events.js

```diff
@@ -25,20 +26,83 @@ dotenv.config();
  * @throws {Error} - A call to next() on an internal server error
  * @description This function creates an event and saves it to the database.
  */
-export const createEvent = async (req, res, next) => {
+
+// export const createEvent = async (req, res, next) => {
+//   try {
+//     const eventData = req.body; // Assuming the event data is sent in the request body
+//     const event = await Event.create(eventData);
+//     if (!event) {
+//       const error = new Error(
+//         "An error occurred while creating the event. Please try again."
+//       );
+//       error.statusCode = 400;
+//       throw error;
+//     }
+//     return res.status(201).json(event);
+//   } catch (error) {
+//     const Event = require("../models/event");
+
+//     return next(error);
+//   }
+// };
+
+// Create a new event
+exports.createEvent = async (req, res) => {
   try {
-    const eventData = req.body; // Assuming the event data is sent in the request body
-    const event = await Event.create(eventData);
-    if (!event) {
-      const error = new Error(
-        "An error occurred while creating the event. Please try again."
-      );
-      error.statusCode = 400;
-      throw error;
-    }
-    return res.status(201).json(event);
+    const { title, description, date, organizer } = req.body;
+
+    const event = new Event({
```

Comment:
This ignores most of the event data  coming from the frontend, like the startDate, endDate, isOnline, price, etc.